VP-5640: Forward the scheme for Linux and non-IIS reverse proxies

trueboroda opened the pull request:
According to doc 
https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-3.1#forward-the-scheme-for-linux-and-non-iis-reverse-proxies
It allow us generate the right links inside the storefront at proxy server usage.

trueboroda committed:
VP-5640: Use forwarded headers

trueboroda committed:
VP-5640: Save requests and responses header to telemetry

trueboroda committed:
VP-5640: Set right order for UseForwardedHeaders() middleware

trueboroda committed:
VP-5640: Final changes for ForwardedHeaders case

asvishnyakov commented on VirtoCommerce.Storefront/Startup.cs:
```diff
@@ -89,6 +91,22 @@ public void ConfigureServices(IServiceCollection services)
 
             services.Configure<StorefrontOptions>(Configuration.GetSection("VirtoCommerce"));
 
+            if (string.Equals(
```
But we already has `ConfigureWebHostDefaults` in Program.cs. I think it should work when `ASPNETCIRE_FORWARDEDHEADERS_ENABLED` specified without this code, out of the box:
https://stackoverflow.com/a/58908072
https://github.com/dotnet/aspnetcore/blob/1480b998660d2f77d0605376eefab6a83474ce07/src/DefaultBuilder/src/WebHost.cs#L240

asvishnyakov commented on VirtoCommerce.Storefront/Startup.cs:
```diff
@@ -89,6 +91,22 @@ public void ConfigureServices(IServiceCollection services)
 
             services.Configure<StorefrontOptions>(Configuration.GetSection("VirtoCommerce"));
 
+            if (string.Equals(
+                Environment.GetEnvironmentVariable("ASPNETCORE_FORWARDEDHEADERS_ENABLED"),
+                "true", StringComparison.OrdinalIgnoreCase))
+            {
+                services.Configure<ForwardedHeadersOptions>(options =>
+                {
+                    options.ForwardedHeaders =
+                        ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
+                    // Only loopback proxies are allowed by default.
+                    // Clear that restriction because forwarders are enabled by explicit 
+                    // configuration.
+                    options.KnownNetworks.Clear();
+                    options.KnownProxies.Clear();
+                });
```
But we already has `ConfigureWebHostDefaults` in Program.cs. I think it should work when `ASPNETCIRE_FORWARDEDHEADERS_ENABLED` specified without this code, out of the box:
https://stackoverflow.com/a/58908072
https://github.com/dotnet/aspnetcore/blob/1480b998660d2f77d0605376eefab6a83474ce07/src/DefaultBuilder/src/WebHost.cs#L240

trueboroda replied:
Will be done.

trueboroda replied:
https://devblogs.microsoft.com/aspnet/forwarded-headers-middleware-updates-in-net-core-3-0-preview-6/

trueboroda committed:
VP-5640: Use ForwardedHeadersOptions out of the box

https://devblogs.microsoft.com/aspnet/forwarded-headers-middleware-updates-in-net-core-3-0-preview-6/

trueboroda committed:
VP-5640: Remove using
